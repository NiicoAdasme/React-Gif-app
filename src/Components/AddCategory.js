import React, { useState } from 'react'
// import GifExpertApp from '../GifExpertApp'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = e => {
        setInputValue(e.target.value);

        // console.log('HandleInputChange llamado')
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        // console.log('HandleSubmit llamado, con valor: ', inputValue);
    
        if(inputValue.trim().length > 2){
            
            setCategories(categ => [inputValue, ...categ]);

            setInputValue('');
        }
    
    };

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input 
                type="text"
                value={inputValue}
                onChange={ handleInputChange }
            />
        </form>  
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};


