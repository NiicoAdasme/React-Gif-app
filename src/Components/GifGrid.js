import React from 'react';
import { useFetchGifs } from '../Hooks/useFetchGifs';
// import { getGifs } from '../Helpers/getGifs';
import { GifGridItem } from './GifGridItem';
// import GifExpertApp from '../GifExpertApp';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

    // const [images, setImages] = useState([]);

    const {data:images, loading} = useFetchGifs(category);

    // getGifs();

    if(category === ""){
        return "Busque una categoria y presione la tecla Enter para buscar el Gif";
    }else{
        return (
            <>
                <h3 className="animate__fadeInDown">Categoria: {category} </h3>
    
                {loading && <p className="animate__flash">Loading...</p>}
    
                <div className="cardGrid">
                {
                    images.map( img => (
                        <GifGridItem 
                            key={img.id}
                            // img={img}
                            {...img} 
                        />
                    ))
                }
            </div>
            </>
            
        );
    }
    
    
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};