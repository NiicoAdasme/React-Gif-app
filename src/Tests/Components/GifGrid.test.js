import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { GifGrid } from "../../Components/GifGrid";
import { useFetchGifs } from "../../Hooks/useFetchGifs";
jest.mock('../../Hooks/useFetchGifs');

describe('Pruebas en el componente GifGrid', () => {

    const category = "Hunter X";
   
    test('Debe de hacer match con el snapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar items cuando se cargan las imagenes useFetchGifs', () => {
       
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/hunterx.jpg',
            title: 'Hunter X gif'
        },
        {
            id: '123',
            url: 'https://localhost/cualquier/hunterx2.jpg',
            title: 'Hunter X gif2'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);

        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    });
    

});
