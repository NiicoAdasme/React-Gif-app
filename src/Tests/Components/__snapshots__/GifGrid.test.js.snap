// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Debe de hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__fadeInDown"
  >
    Categoria: 
    Hunter X
     
  </h3>
  <p
    className="animate__flash"
  >
    Loading...
  </p>
  <div
    className="cardGrid"
  />
</Fragment>
`;
